### ISSUE TITLE
[Documentation Request] A guide on how to create dictionaries

### ISSUE THREAD

**Issue by wildwestrom** [FooSoft/yomichan#2206]
Here's the situation.

I have some XML files with dictionary data not available for yomichan. I'd like to take that data and turn it into JSON data that can be read by yomichan. I found that there are various JSON schemas. I can also deduce what some of the fields mean, while others (e.g [the `format` enum](https://github.com/FooSoft/yomichan/blob/6694a9d9c3a6770ee744310056cadd3567386f3f/ext/data/schemas/dictionary-index-schema.json#L23-L27)) are not obviously documented.

If somebody understands the internals better, I think having documentation for creating dictionaries would be a great boon to the users and developers.

**Comment by MarvNC** [comment]
I was wondering the same thing. It seems `format` is usually 3 but I have no idea why; maybe it's the version of the schema?
In the [tag bank](https://github.com/FooSoft/yomichan/blob/6694a9d9c3a6770ee744310056cadd3567386f3f/ext/data/schemas/dictionary-tag-bank-v3-schema.json), I was wondering what the second term, category is. It seems it might be better as an enum with some documentation? Looking at KANJIDIC, there are `frequent`, `misc`, `code`, `class`. and `index`. I think they're hard coded [here](https://github.com/FooSoft/yomichan/blob/331a2e62941e04a4d50a21faefed663a92ddc00a/ext/js/display/display-generator.js#L156) with the exception of `frequent`, which I assume is for tags.

**Comment by toasted-nutbread** [comment]
You're right, it's not a well documented process at this point, aside from some stuff in the schemas. Something that I often find helpful is extracting some of the dictionaries included on the Yomichan homepage and to use them as reference.

`format` should be `3` for all new dictionaries. The other versions correspond to legacy dictionary formats which supported less information than the current ones.

Tag banks: `category` is somewhat arbitrary, but it's kind of used for two purposes:
* A namespace, which allows disambiguation between tags with the same short label but potentially different description.
* For styling purposes, see [display.css](https://github.com/FooSoft/yomichan/blob/c331ceae997a49e3c7125853c2225964dc17749b/ext/css/display.css#L760-L789).

Tag banks: (popularity) `score` is used for sorting tags, and the numbering is also somewhat arbitrary. There is a similar `score` value used for definitions. Higher score tags will show up earlier in the tag list, lower score tags show up later. There is a similar sorting based on score for definitions.
